File: codigo/soluciones/kat/baloni.py
N = int(input())
H = list(map(int, input().split()))
mx = max(H)
arrow = [0 for _ in range(mx + 1)]
ans = 0
for x in H:
    if arrow[x] == 0:
        ans += 1
    else:
        arrow[x] -= 1
    arrow[x - 1] += 1
print(ans)

File: codigo/soluciones/kat/baloni.java
import java.io.*;
import java.util.*;
class Baloni {
    private void _main() {
        Scanner sc = new Scanner(new BufferedInputStream(System.in));
        int N = sc.nextInt();
        int[] H = new int[N];
        int mx = 0;
        for (int i = 0; i < N; ++i) {
            H[i] = sc.nextInt();
            mx = Math.max(mx, H[i]);
        }
        int[] arrow = new int[mx + 1];
        for (int i = 0; i < mx + 1; ++i) {
            arrow[i] = 0;
        }
        int ans = 0;
        for (int i = 0; i < N; ++i) {
            if (arrow[H[i]] == 0) {
                ++ans;
            }else {
                --arrow[H[i]];
            }
            ++arrow[H[i] - 1];
        }
        System.out.println(ans);
        sc.close();
    }
    public static void main(String[] args) {
        (new Baloni())._main();
    }
}
